### PR TITLE
DAT-7316, DAT-7343 & validation

### PR DESCRIPTION
**Pull request by Tabuu**
This pull request should:
- Prevent attachments from not being downloaded from APPEEE and uploaded to ArcGIS.
- Add a cloud function to restore the, in the past, not downloaded attachments.
- Resolve the issue of "foto_bop_att_id field" not being filled in. Which was likely caused by the attachment missing.

### CODE REVIEW

File: functions/sync_images/main.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    :param request: The request object to extract from.
+    :type request: flask.Request
+
+    :return: A dictionary of all arguments.
+    :rtype: dict
+    """
+
+    arguments = dict()
+    if not request:
+        return arguments
```
Comment (maljabary) [inline functions/sync_images/main.py:132]: one place to return the expected value of a function is recommended. why not use:

if request: and remove the return

File: functions/sync_images/main.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    # Update arcgis entry even when no attachment was downloaded.
+    force_arcgis_update = arguments.get("force_arcgis_update", False)
+
+    # Retry options
+    request_retry_options = arguments.get("request_retry_options", {})
+
+    storage_client = storage.Client()
+    bucket = storage_client.get_bucket(IMAGE_STORE_BUCKET)
+
+    attachment_service = AttachmentService(storage_client, get_requests_session(**request_retry_options))
```
Comment (maljabary) [inline functions/sync_images/main.py:53]: abstraction is recommended. why not hide get_requests_session within the functions, so users of that functions don't have to worry about it when calling the function.

File: functions/sync_images/main.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    publish_service = PublishService(TOPIC_NAME, get_requests_session(**request_retry_options))
```
Comment (maljabary) [inline functions/sync_images/main.py:54]: same here

File: functions/sync_images/form_object.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+                        f"{self.attachment_bucket_base_path}{survey_value}",
+                        f"{self.attachment_download_base_url}/{survey_field}_{survey_value}"
+                    )
+
+                    attachments.append(attachment)
+
+        return attachments
+
+
+def _is_survey_value_attachment(value):
```
Comment (maljabary) [inline functions/sync_images/form_object.py:97]: Why this function outside the class?

File: functions/sync_images/form_object.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    return isinstance(value, str) and path.splitext(value)[1] in IMAGE_FILE_EXTENSIONS
```
Comment (maljabary) [inline functions/sync_images/form_object.py:98]: what about when index 1 non existing in splitext for some reason?

File: functions/sync_images/coordinate_service.py [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+        form_sleutel = get_from_dict(form_entry, key_field_mapping.split("/"))
+        if not form_sleutel:
+            logging.info(
+                f"Feature does not contain '{key_field_mapping}' field, skipping this."
+            )
+            return None
+
+        address = form_sleutel.split("_")
+
+        address_postcode = address[0]
```
Comment (maljabary) [inline functions/sync_images/coordinate_service.py:94]: what if address doesn't have index 0?

File: functions/sync_images/constant.py [hunk 7]
```diff
@@ -0,0 +1,6 @@
+PROVIDER_ID_KEY = "ProviderId"
```
Comment (maljabary) [inline functions/sync_images/constant.py:1]: code duplicate: https://recognize.atlassian.net/browse/DAT-7355

**Review by maljabary — APPROVED** [review]

File: functions/sync_images/form_object.py (hunk 5 above)
Comment (Tabuu) [reply functions/sync_images/form_object.py:98]: Index 1 always exists [by definition](https://docs.python.org/3/library/os.path.html#os.path.splitext), however I will refactor the code to be more explicit.

**Pull request merged**